src/boardlib/api/aurora.py

```diff
@@ -391,3 +392,204 @@ def save_climb(
+    
+    for raw_entry in raw_entries:
+        if db_path:
+            climb_name = get_climb_name_from_db(db_path, raw_entry["climb_uuid"])
+        else:
+            climb_name = get_climb_name(board, raw_entry["climb_uuid"])
+        
+        yield {
+            "uuid": raw_entry["uuid"],
+            "user_id": raw_entry["user_id"],
+            "climb_uuid": raw_entry["climb_uuid"],
+            "climb_name": climb_name if climb_name else "Unknown Climb",
+            "angle": raw_entry["angle"],
+            "is_mirror": raw_entry["is_mirror"],
+            "bid_count": raw_entry["bid_count"],
+            "comment": raw_entry["comment"],
+            "climbed_at": raw_entry["climbed_at"],
+            "created_at": raw_entry["created_at"],
+        }
+
+def get_displayed_grade_from_db(database, climb_uuid, angle, grades_dict):
+    conn = sqlite3.connect(database)
+    cursor = conn.cursor()
+    cursor.execute(
+        "SELECT display_difficulty FROM climb_stats WHERE climb_uuid = ? AND angle = ?", 
+        (climb_uuid, angle)
+    )
+    row = cursor.fetchone()
+    conn.close()
+    if row:
+        difficulty_value = round(row[0])
+        grade_info = grades_dict.get(difficulty_value, {})
+        return grade_info.get("french_name", "NA")
+    return "NA"
+
+def get_full_logbook_entries(board, username, password, grade_type="font", db_path=None):
+    # Get bids and ascents data
+    login_info = login(board, username, password)
+    token = login_info["token"]
+    user_id = login_info["user_id"]
+    
+    bids_entries = list(bids_logbook_entries(board, username, password, db_path))
+    raw_ascents_entries = get_logbook(board, token, user_id)
+    
+    # Convert bids entries to DataFrame
+    bids_df = pd.DataFrame(bids_entries)
+    bids_df['climbed_at'] = pd.to_datetime(bids_df['climbed_at'])
+    
+    # Process raw ascents entries and convert to DataFrame
+    ascents_entries = []
+    grades = get_grades(board)
+
+    # Convert grades to dictionary for easy lookup
+    grades_dict = {grade['difficulty']: grade for grade in grades}
+
+    for raw_entry in raw_ascents_entries:
+        if not raw_entry["is_listed"]:
+            continue
+        if db_path:
+            climb_name = get_climb_name_from_db(db_path, raw_entry["climb_uuid"])
+            displayed_grade = get_displayed_grade_from_db(db_path, raw_entry["climb_uuid"], raw_entry["angle"], grades_dict)
+        else:
+            climb_name = get_climb_name(board, raw_entry["climb_uuid"])
+            displayed_grade = "NA"
+        
+        grade_info = grades_dict.get(raw_entry["difficulty"], {})
+        logged_grade = grade_info.get("french_name" if grade_type == "font" else "verm_name", "Unknown")
+        
+        ascents_entries.append({
+            "board": board,
+            "angle": raw_entry["angle"],
+            "climb_uuid": raw_entry["climb_uuid"],
+            "name": climb_name if climb_name else "Unknown Climb",
```

Comment:
When is "Unknown Climb" occurring? If this isn't something that we expect to happen normally, I'd prefer to just leave it as is. It seems like we should raise errors if we can't find the climb name.

Reply:
agreed and changed.